src/RandN/Extensions/DistributionExtensions.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace RandN.Extensions
+{
+    /// <summary>
+    /// Contains extension methods for the <see cref="IDistribution{TResult}"/> interface.
+    /// </summary>
+    public static class DistributionExtensions
+    {
+        /// <summary>
+        /// Transforms a distribution by mapping its values using the selector provided.
+        /// This method implements the "map" operator from functional programming principles.
+        /// </summary>
+        /// <example>
+        /// This method enables use of LINQ query syntax over <see cref="IDistribution{TResult}"/>. For example:
+        /// <code>
+        /// IDistribution&lt;int&gt; evenNumbersDistribution =
+        ///     from i in Uniform.NewInclusive(0, 5)
+        ///     select i * 2;
+        /// </code>
+        /// </example>
+        /// <typeparam name="TSource">The generic type of the input distribution.</typeparam>
+        /// <typeparam name="TResult">The generic type of the output distribution.</typeparam>
+        /// <param name="distribution">The distribution to be transformed.</param>
+        /// <param name="selector">The projection to be applied to values from the distribution.</param>
+        public static IDistribution<TResult> Select<TSource, TResult>(
```

Comment:
Is there any reason this can't return the `SelectDistribution` type outright? There are potential (though minor) performance benefits since methods calls won't need to be a virtual. The caller could always cast it to an `IDistribution<TResult>` later if they needed to.

Reply:
Currently the only reason why this is needed is that the `SelectDistribution` type is private, but I could easily make it public (maybe with an internal constructor).

One benefit of returning the interface rather than the concrete type is that the caller probably only cares about getting an object which implements the interface. I think that's why the standard LINQ `Select()` returns an `IEnumerable<TResult>`. I don't know how much of a big deal that is though.

The benefit of returning the concrete type is, as you say, improved performance. There's also the fact that most other methods in this library return the concrete type, so consistency with that would be good.

TL;DR I'm happy to return the concrete type if you think it's best!

Reply:
Just tried it out locally. One oddity is that the public interface of `SelectDistribution<TSource, TResult>` only cares about one of the type parameters; namely `TResult`. Not sure if that sways things or not.

Reply:
Those are good points - I was thinking of how ImmutableArray has its own extension methods that are optimized for it instead of the standard LINQ expressions. But I don't think that's relevant here since this would be the generalized method. So I'm leaning back towards returning IDistribution<TResult> again.